src/main/java/hudson/plugins/audit_trail/AuditTrailPlugin.java

```diff
@@ -24,158 +24,146 @@
-      + "cancelQueue|stop|toggleLogKeep|doWipeOutWorkspace|createItem|createView|toggleOffline|"
-      + "cancelQuietDown|quietDown|restart|exit|safeExit)";
+            + "cancelQueue|stop|toggleLogKeep|doWipeOutWorkspace|createItem|createView|toggleOffline|"
+            + "cancelQuietDown|quietDown|restart|exit|safeExit)";
     private boolean logBuildCause = true;
 
-    private List<AuditLogger> loggers = new ArrayList<AuditLogger>();
+    private List<AuditLogger> loggers = new ArrayList<>();
 
     private transient boolean started;
 
     private transient String log;
-    private transient int limit = 1, count = 1;
 
     public String getPattern() { return pattern; }
     public boolean getLogBuildCause() { return logBuildCause; }
     public List<AuditLogger> getLoggers() { return loggers; }
 
-    @Override public void start() throws Exception {
-        // Set a default value; will be overridden by load() once customized:
+    public AuditTrailPlugin() {
         load();
-        applySettings();
-
-        // Add Filter to watch all requests and log matching ones
-        PluginServletFilter.addFilter(new AuditTrailFilter(this));
     }
 
-    @Override public void configure(StaplerRequest req, JSONObject formData)
-            throws IOException, ServletException, FormException {
-        pattern = formData.optString("pattern");
-        logBuildCause = formData.optBoolean("logBuildCause", true);
+    @Override
+    public boolean configure(StaplerRequest req, JSONObject formData) {
         // readResolve makes sure loggers is initialized, so it should never be null.
+        // TODO this should probably be moved somewhere else
         loggers.forEach(AuditLogger::cleanUp);
-        loggers = Descriptor.newInstancesFromHeteroList(
-                req, formData, "loggers", getLoggerDescriptors());
-        save();
+        req.bindJSON(this, formData);
         applySettings();
+        return true;
     }
 
+    @DataBoundSetter
+    public void setPattern(String pattern) {
+        this.pattern = Optional.ofNullable(pattern).orElse("");
+        save();
+    }
+
+    @DataBoundSetter
+    public void setLogBuildCause(boolean logBuildCause) {
+        this.logBuildCause = logBuildCause;
+        save();
+    }
+
+    /**
+     * @deprecated as of 2.6
+     **/
+    @Deprecated
     public DescriptorExtensionList<AuditLogger, Descriptor<AuditLogger>> getLoggerDescriptors() {
-        return Jenkins.getInstance().getDescriptorList(AuditLogger.class);
+        return AuditLogger.all();
     }
 
+    @DataBoundSetter
+    public void setLoggers(List<AuditLogger> loggers) {
+        this.loggers = Optional.ofNullable(loggers).orElse(Collections.emptyList());
+    }
 
+    @PostConstruct
     private void applySettings() {
         try {
             AuditTrailFilter.setPattern(pattern);
+        } catch (PatternSyntaxException ex) {
+            ex.printStackTrace();
```

Comment:
By doing so, you are enforcing the stack trace to be log in System.err. It's preferable to either not do it completely or to log using the regular LOGGER.